research/query_service/ir/graph_proxy/src/apis/write_graph.rs

```diff
@@ -13,22 +13,24 @@
 //! See the License for the specific language governing permissions and
 //! limitations under the License.
 
-use crate::apis::{Edge, Vertex};
+use ir_common::NameOrId;
+
+use crate::apis::graph::PK;
+use crate::apis::DynDetails;
 use crate::GraphProxyResult;
 
 /// The interfaces of writing data (vertices, edges and their properties) into a graph.
 pub trait WriteGraphProxy: Send + Sync {
     /// Add a vertex
-    fn add_vertex(&mut self, vertex: Vertex) -> GraphProxyResult<()>;
-
-    /// Add a batch of vertices
-    fn add_vertices(&mut self, vertices: Vec<Vertex>) -> GraphProxyResult<()>;
+    fn add_vertex(
+        &mut self, vertex_pk: PK, label: &NameOrId, properties: Option<&DynDetails>,
```

Comment:
`properties` better be `Option<DynDetails>`

Reply:
Done.